### PR TITLE
Added warning for nan values when generating inputs

### LINKED ISSUE #278: api.generate_inputs() could issue warnings when writing NaN values
When `api.generate_inputs()` is used with a data source, it could be useful to issue warnings for each variable that gets a NaN value in the written file, rather that expecting user to go and check for himself.

But without a data source, it is rather obvious that it will contain NaN values, so I would not change the behavior in this case.

### PR DESCRIPTION
**Pull request by ScottDelbecq**
This PR closes #278 by adding a warning when NaN values are encountered in the source file of `generate_inputs`.

### CODE REVIEW

**Review by christophe-david — CHANGES_REQUESTED** [review]
Thank you for this work, and for having tried the caplog fixture.

Some issues to fix and it will be right.

File: src/fastoad/io/configuration/configuration.py [hunk 1]
```diff
@@ -230,8 +231,15 @@ def write_needed_inputs(
         if source_file_path:
             ref_vars = DataFile(source_file_path, formatter=source_formatter)
             variables.update(ref_vars, add_variables=False)
+            nan_variable_names = []
             for var in variables:
                 var.is_input = True
+                # Checking if variables have NaN values
+                if np.all(np.isnan(var.value)):
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad/io/configuration/configuration.py]: I think there is very little chances that this occurs but you are right!
Corrected in 8912826.
I also corrected for the verification during computation in 278e90a.

File: src/fastoad/io/configuration/configuration.py [hunk 2]
```diff
@@ -230,8 +231,15 @@ def write_needed_inputs(
...
+                    nan_variable_names.append(var.name)
+            if nan_variable_names:
+                msg = "The following variables have NaN values: %s" % nan_variable_names
+                _LOGGER.warning(msg)
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad/io/configuration/configuration.py]: Ok. Changed in c2d6d3c.

File: src/fastoad/io/configuration/tests/test_configuration.py [hunk 3]
```diff
@@ -124,7 +124,15 @@ def test_problem_definition_with_xml_ref(cleanup):
         result_folder_path = pth.join(RESULTS_FOLDER_PATH, "problem_definition_with_xml_ref")
         conf.input_file_path = pth.join(result_folder_path, "inputs.xml")
         conf.output_file_path = pth.join(result_folder_path, "outputs.xml")
+        ref_input_data_path_with_nan = pth.join(DATA_FOLDER_PATH, "ref_inputs_with_nan.xml")
         ref_input_data_path = pth.join(DATA_FOLDER_PATH, "ref_inputs.xml")
+
+        # Test that the presence of NaN values in inputs logs a warning
+        conf.write_needed_inputs(ref_input_data_path_with_nan)
+        for record in caplog.records:
+            assert record.levelname == "WARNING"
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad/io/configuration/tests/test_configuration.py]: Changed in ee45d1e.

File: src/fastoad/io/configuration/tests/data/ref_inputs_with_nan.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  ~ along with this program.  If not, see <https://www.gnu.org/licenses/>.
+  -->
+
+<aircraft>
+    <x>nan</x>
+    <z units="m**2">5.0 2.0</z>
+    <y1>nan</y1><!-- If this variable is kept in the input IVC, OpenMDAO will raise an exception because it is an output of a component
+                     As it should be ignored, its NaN value should bring no harm.-->
+    <foo>nan</foo><!-- This totally unused variable is expected to be transferred in output file -->
+    <bar>42</bar><!-- This totally unused variable is expected to be transferred in output file -->
```
[the remark this answers is not in the document]
Comment (ScottDelbecq) [reply src/fastoad/io/configuration/tests/data/ref_inputs_with_nan.xml]: Corrected in 37b3aac.

File: src/fastoad/io/configuration/configuration.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (christophe-david) [reply src/fastoad/io/configuration/configuration.py]: Nice. Thank you.

**Review by christophe-david — APPROVED** [review]
All right !

**Pull request merged**